lib/phase/merge.js

```diff
@@ -79,6 +79,32 @@ function mergeSharedConfig(parsedDoc, pipelineTemplate) {
         newPipeline.shared.image = mergedImage;
     }
 
+    if (parsedDoc.cache) {
```

Comment:
We may have to concatenate the cache paths from the pipeline configuration with cache paths inherited from the pipeline template for all the scopes (`pipeline`, `event`, `jobs`).
If we do high level replace, the cache paths configured for pipeline template jobs will be lost.